### ISSUE TITLE
(Feature Request) Dedicated Password Manager button on Action Bar

### ISSUE THREAD

**Issue by m-blodwyn** [futo-org/android-keyboard#325]
Normally the intended behaviour of password manager suggested logins appearing on the Action Bar work as intended for me, however some websites just seem to be poorly designed no matter what keyboard app I use and don't trigger the password manager prompts (for example, Twitter on Firefox never suggests my emails to sign in for me). A dedicated button in the extended Action Bar menu to open my chosen password manager application (in my case, Proton Pass) would be extremely helpful if possible. It also would help alleviate any relience on fixing specific sites' input fields etc. by allowing me to open my password manager to grab the info I want wherever I choose to.

**Comment by stevetheman1031** [comment]
I came here to comment the same thing! I second this and I use bitwarden (selfhosted). The drawover other apps usually work but unless the app stays on the password prompt in the background (some don't like samsung secure folder) it works fine. Before I switched to the FUTO keyboard, the Samsung keyboard showed the option to autofill from bitwarden.

**Comment by GOinfo-Ltd** [comment]
I second this as well. However, I'd go a step further and suggest to not only support the default password manager, but support secondary ones as well.

This is a feature that is lacking on Android, but I am personally using two different password managers (one for work, one for private use), and you can not easily switch between the two.

Having a list of password managers to choose from would be an incredible help to the (admittedly few) people in my usecase.

Thank you in advance!

**Comment by weedy** [comment]
I'm on android 10 so I don't have the built-in, the integrated in-line password manager autofill support. 

So a way to kick bitwarden to suggest me a password would be appreciated.

**Comment by 32bitx64bit** [comment]
Same issue here. Bitwarden support would be awesome.

**Comment by GameTec-live** [comment]
A way to tell the andoroid password autofill service (or whatever the service is called where you select your preferred manager in the settings) to just pop open suggestions NOW would be amazing

**Comment by VoxelPrismatic** [comment]
My issue is that most websites now do a second page for the password, which always seems to cause trouble with Proton Pass. About a 50% chance it fails, so a button to open Proton Pass in the event it does fail would be handy.

**Comment by primaleph** [comment]
I use 1Password for work. So far, I haven't been able to get it to play nice with this keyboard. The built-in password manager is not good enough. So I really need this.

**Comment by cco** [comment]
> I use 1Password for work. So far, I haven't been able to get it to play nice with this keyboard. The built-in password manager is not good enough. So I really need this.

Same here as well. I pay for 1password and was disappointed that I couldn't get it working with the keyboard. Not sure if this issue applies to other managers but it's quite a headache to not have it.

**Comment by abb128** [comment]
@primaleph @cco 
1Password should work fine with the keyboard. Did you set it up correctly for inline autofill? Go to 1password > Settings > Autofill and make sure everything is enabled, "Display Autofill prompts" is set to "Above supported keyboards". Then go to Android Settings > Passwords, passkeys & accounts > ensure Preferred service is set to 1Password. Then make sure you have an entry that is valid (e.g. add a login for https://github.com/login with a test username and password) and try navigating to that page in your browser and highlighting the username or password field. On my side with this configuration, the inline autofill appears just fine

**Comment by VoxelPrismatic** [comment]
I would still prefer a dedicated button, because with some apps, the username autofill is fine, but not the password, which means I have to exit the app, go all the way to my password manager, copy the password, and then come back.

This can be a simple shortcut similar to the new word styles feature.

**Comment by primaleph** [comment]
Yes, 1Password is set up correctly. I know this because I have worked in
tech support for 28 years, and because it works perfectly with every other
keyboard I've used. With Futo, sometimes it works and other times it acts
as if it's not there at all.

Two different users have reported this issue. Maybe you should consider
that this is a bug on your end, rather than blaming it on those users. I
found your response condescending, and honestly a little rude. This is not
a professional way to respond to a bug report.

**Comment by cco** [comment]
> @primaleph @cco 
> 1Password should work fine with the keyboard. Did you set it up correctly for inline autofill? Go to 1password > Settings > Autofill and make sure everything is enabled, "Display Autofill prompts" is set to "Above supported keyboards". Then go to Android Settings > Passwords, passkeys & accounts > ensure Preferred service is set to 1Password. Then make sure you have an entry that is valid (e.g. add a login for https://github.com/login with a test username and password) and try navigating to that page in your browser and highlighting the username or password field. On my side with this configuration, the inline autofill appears just fine

Thanks for the rubber duck. It looks like my inline autofill for futo had been disabled, preventing 1password from displaying suggestions. This really ought to be a default feature and not something a user needs to look for.

**Comment by abb128** [comment]
@cco If you're referring to the inline autofill toggle in FUTO keyboard settings, that option is already on by default. It shouldn't be off unless you manually turned it off in the past or by accident

**Comment by Webbeh** [comment]
Irrelevant of the people being annoying in here, I'd definitely appreciate having a dedicated button as well for the few apps that don't trigger the password search at all or do them incorrectly.

**Comment by weedy** [comment]
Bitwarden gives me a quick tile in the notification shade.
I use it CONSTANTLY with homedepot app and website.

Just as I have added shortcut for select all, cut, and clipboard history I'd love to force inline autofill or even just launch password manager. Actually launching manager might be better since payment and ID autofill never works for me.

**Comment by DavidGameDev** [comment]
+1 from me. Just a button to call Autofill when it's not automatically suggested

**Comment by SailLively** [comment]
I would love to be able to launch my password manager as a pinned action or one of the actions when I hold on the enter key, I think that integration is easy enough, or even an action to simply open any app of the persons choosing, wouldn't that be cool!

**Comment by Badwai** [comment]
Just wanted to bump this issue again. The inconsistency of android apps triggering the password manager is incredibly frustrating. It would be an immense benefit to have an action available to manually activate the process. At a basic level it seems like it should be pretty straight forward. An action which immediately activates the system default password manager.

**Comment by juke32** [comment]
Your comment sent me an email update, I got excited that someone may have coded it in. anyway..

I was able to add the action! Yes it is vibe coded, but it is simple
- AndroidManifest.xml `+1ish` line exposes default autofill service
- Registry.kt  `+1` line to register the action
- strings-uix.xml  `+1` line for the button label
- PasswordManagerAction.kt - new added action that opens the default auto fill service

Only reason this could be easily done was because FUTO made the organization aspect easy. But the actual file that handles the button may need to be changed and reworked by someone smarter than me+clanker

I made the workflow of building it easy through github actions so someone could easily tweak it untill it works for a lot of people, then the feature can be hopefully be implemented in the main app!! I am willing to take on the task of vetting this small feature but haven't done something like this

Check it out! (can be installed without affecting official FUTO-Keyboard app)
https://github.com/juke32/futo-keyboard-juke32

<img width="227" height="442" alt="Image" src="https://github.com/user-attachments/assets/c8b0903a-f5dd-4083-8349-f8678bbac1b9" />

**Comment by chrschorn** [comment]
@juke32 Do you want to submit a PR? It might help put this on the maintainers' radar!

It looks like autofill on Android is just [generally flaky](https://9to5google.com/2024/10/06/android-autofill-password-manager-problems/). I often have issues with password managers not getting triggered on any Android keyboard - seems I'm not alone. It would be amazing if Futo Keyboard could trigger the Android autofill more often / in more cases. A dedicated button [like GBoard has for Google password manager](https://9to5google.com/2025/08/12/gboard-autofill-with-google/) would be ideal to work around the Android issues (which Android seems to refuse to fix).

**Comment by bywilfried** [comment]
@juke32 thank you for that. 
My concern is, when we select password manager, it just open it, it doesn't automatically open as if we where on a page we could autofill, and so we can't just clic on the pass we want to use to run the autofill, we need to copy/past :(

**Comment by abb128** [comment]
@bywilfried That would not be possible as far as I'm aware unfortunately. There is no way for an IME in Android to trigger in-app autofill. The most it can do is open the password manager as if you launched it from your home screen, and that's why this isn't implemented in FUTO Keyboard since it feels somewhat buggy/broken

**Comment by SailLively** [comment]
@abb128 thoughts on an action to open any defined app the user sets, I think people would find that helpful, and it would be easy to maintain and setup. Could be used for a to-do/ notetaking app, messaging app, 2fa app, password manager, anything else someone uses often from the keyboard

**Comment by juke32** [comment]
@chrschorn I experienced the flakey-ness. Testing for each manager would be a pain and that work would basically use the same service the keyboard could already use to improve the current autofill...

I don't understand the G-board implementation, but the shortcut only seems to work with the Google Password Manager

After playing around I think it is best to:
1. help fix the current autofill (even if that means going to the Android OSP)
2. use workarounds like accesibility service to see open websites and serve autofill on shortcut or on field press?
3. Ability to add applicaitons as actions. Maybe even implement application shortcuts?
